scw/locality.go

```diff
@@ -45,6 +49,17 @@ func (zone *Zone) String() string {
 	return string(*zone)
 }
 
+// Region returns the parent Region for the Zone.
+// Manipulates the string directly to allow unlisted zones formatted as xx-yyy-z.
+func (zone *Zone) Region() (Region, error) {
```

Comment:
What do you think about

```go
func (zone *Zone) Region() (Region, error) {
    zoneStr := zone.String()
    if (validate.IsZone(zoneStr)) {
        return Region(zoneStr[:len(zoneStr)-2])
    }
    return "", fmt.Errorf("invalid zone '%v'", zoneStr)
}
```

Reply:
Using `IsZone()` is a good idea.

`zoneStr[:len(zoneStr)-2] ` could fail for `fr-par-12`

I prefer to work with separators than length.

Reply:
Nop, `IsZone("fr-par-12") == false`.
Hint: https://github.com/scaleway/scaleway-sdk-go/blob/master/validation/is.go#L12

Reply:
I'm ok with using the regex defined in validation.go to parse the Region from the zone if necessary. 

I don't think we should work with the length.

`2` looks a bit like a magic number

More generally, I think we should work with regex, then separators, then length if nothing else is available : I find it easier to read and it's more robust.